ElementX/Sources/Screens/HomeScreen/HomeScreenViewModel.swift

```diff
@@ -235,24 +239,34 @@ class HomeScreenViewModel: HomeScreenViewModelType, HomeScreenViewModelProtocol
         }
         
         state.rooms = rooms
+        
+        MXLog.info("Finished updating rooms")
     }
     
     private func buildRoom(with details: RoomSummaryDetails) -> HomeScreenRoom {
-        let avatarImage = details.avatarURL.flatMap { userSession.mediaProvider.imageFromURL($0, avatarSize: .room(on: .home)) }
+        var room: HomeScreenRoom! = roomsForIdentifiers[details.id]
+        
+        if room == nil {
+            room = HomeScreenRoom(id: details.id,
+                                  roomId: details.id,
+                                  avatarURL: details.avatarURL)
+        }
```

Comment:
Totally unimportant, but this is simpler (although it looks a mess with GH formatting)
```suggestion
        var room: HomeScreenRoom! = roomsForIdentifiers[details.id] ?? HomeScreenRoom(id: details.id,
                                                                                      roomId: details.id,
                                                                                      avatarURL: details.avatarURL)
```

Reply:
I think I prefer mine better as it's more in your face